wallet/wallet.go

```diff
@@ -4314,13 +4314,27 @@ func (w *Wallet) PublishTransaction(tx *wire.MsgTx, serializedTx []byte, n Netwo
 
 	err = n.PublishTransactions(context.TODO(), tx)
 	if err != nil {
-		if relevant {
-			if err := w.PurgeUnminedTransaction(&txHash); err != nil {
-				log.Warnf("Failed to purge added unmined transaction: %v", err)
+		// Return the error and purge relevant txs from the store only if the
+		// backend returns an error different than "already have transaction".
+
+		// Unwrap to access the underlying RPC error code.
+		for {
+			if e, ok := err.(*errors.Error); ok && e.Err != nil {
+				err = e.Err
```

Comment:
unwrapping the error and reassigning err this way would strip any additional details added to the error from this method's return.  You could probably declare a *dcrjson.RPCError above and set it from inside here, checking if it's not nil below.

Reply:
One alternative that I thought about that I'd like to run past you: What if the backends detected this particular condition (`ErrRPCDuplicateTx`) and, when creating the actual `errors.Error` object, added a `KindExists`? Would avoid having to manually unwrap the error every time.

Reply:
That's also not a bad idea